delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceUris.scala

```diff
@@ -0,0 +1,175 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, JsonLdContext}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.AclAddress
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{ApiMappings, ProjectBase, ProjectRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+
+/**
+  * Holds information about the different access Uri of a resource
+  */
+sealed trait ResourceUris extends Product with Serializable {
+
+  /**
+    * @return the relative access [[Uri]]
+    */
+  private[sdk] def relativeAccessUri: Uri
+
+  /**
+    * @return the relative access [[Uri]] in a short form
+    */
+  private[sdk] def relativeAccessUriShortForm: Uri
+
+  /**
+    * @return the access [[Uri]]
+    */
+  def accessUri(implicit base: BaseUri): Uri =
+    relativeAccessUri.resolvedAgainst(base.endpoint.finalSlash())
+
+  /**
+    * @return the access [[Uri]] in a short form
+    */
+  def accessUriShortForm(implicit base: BaseUri): Uri =
+    relativeAccessUriShortForm.resolvedAgainst(base.endpoint.finalSlash())
+
+  /**
+    * @return the incoming [[Uri]]
+    */
+  def incoming(implicit base: BaseUri): Option[Uri]
+
+  /**
+    * @return the outgoing [[Uri]]
+    */
+  def outgoing(implicit base: BaseUri): Option[Uri]
+
+  /**
+    * @return the incoming [[Uri]] in a short form
+    */
+  def incomingShortForm(implicit base: BaseUri): Option[Uri]
+
+  /**
+    * @return the outgoing [[Uri]] in a short form
+    */
+  def outgoingShortForm(implicit base: BaseUri): Option[Uri]
+
+}
+
+object ResourceUris {
+
+  final private case class WithNavigation private (relativeAccessUri: Uri, relativeAccessUriShortForm: Uri)
+      extends ResourceUris {
+    override def incoming(implicit base: BaseUri): Option[Uri]          = Some(accessUri / "incoming")
```

Comment:
We only use these links during ResourceF serialization no ?
We could have an encoder taking a BaseUri and generate the JsonObject with the applicable links for each one.
A bit like we do, for subjects in ResourceF

Reply:
`ResourceUris` has nothing to do with serialization. As the class says
> A collection of resource uris

Reply:
incoming and outgoing links are also used and stored in the elasticsearch plugin